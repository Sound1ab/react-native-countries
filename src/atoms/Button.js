import styled from 'styled-components';
import {StyleSheet, Text, View, TouchableOpacity} from 'react-native';

const Button = styled.TouchableOpacity`
    padding: 16px 40px;
    background-color: tomato;
    border-radius: 5px;
`;

export default Button;