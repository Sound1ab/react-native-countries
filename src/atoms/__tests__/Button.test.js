import React from 'react';
import Button from '../Button.js';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('Button', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<Button>Button text</Button>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
