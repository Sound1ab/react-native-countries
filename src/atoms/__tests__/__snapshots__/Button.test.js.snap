// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "tomato",
      "borderRadius": 5,
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 40,
      "paddingRight": 40,
      "paddingTop": 16,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  Button text
</View>
`;
